signer/src/stacks/api.rs

```diff
@@ -320,16 +321,109 @@ impl TenureBlocks {
     /// # Note
     ///
     /// The struct doesn't need to contain all the blocks in a tenure.
+    #[cfg(any(test, feature = "testing"))]
     pub fn into_blocks(self) -> Vec<NakamotoBlock> {
         self.blocks
     }
+}
 
-    /// Return an iterator of Stacks blocks included in this object.
-    pub fn as_stacks_blocks(&self) -> impl Iterator<Item = StacksBlock> + '_ {
-        let bitcoin_anchor = &self.anchor_block_hash;
-        self.blocks
-            .iter()
-            .map(|block| StacksBlock::from_nakamoto_block(block, bitcoin_anchor))
+/// A slimmed down [`NakamotoBlockHeader`]
+#[derive(Debug, Clone, PartialEq)]
+pub struct StacksBlockHeader {
+    /// The total number of StacksBlocks and NakamotoBlocks preceding this
+    /// block in this block's history.
+    pub block_height: u64,
```

Comment:
We can use `model::StacksBlockHeight` now!

Reply:
Good call. I'll get to this when I change the base branch of the PR and merge main.